### PR TITLE
authentication-demo

### PR DESCRIPTION
**Pull request by gradygabriel10**
## Summary
Created the authentication system (using my own login page template). Once, the front-end team has done with their template, we can start implementing the functions to the page.  Functions that are implemented are:
1. Register Button 
2. Login Button
3. Google Sign in Button
4. Forgot Password --> Send Instructions Button

Important Notes:
Users couldn't change their password if they sign in with Goggle.

### Checklist
- [ ] Added new tests (You know the code best to make new tests)
- [x] Resolved merge conflicts (Prefer merge over rebase)
- [ ] Status check passed (Linted and tested)
- [ ] Reviewed (At least one reviewer's approval)

## Changelog/Brief Description of Solution

[TYPE] - Message

## Test Plan/Special Instructions for Team
If team members want to try the demo, make sure these are included. 
1. Set Real-Time Database Rules (.read .write) to true so that the data is allowed to be read/written. 
2. Add local domain (Authentication -> Sign-in method -> Authorized domains). For example: 127.0.0.1
3. Open login.html with the Live Server 
4. Once the user is registered using the page, verify the email using the link inside the user's inbox email. 
5. Check inside Firebase (Authentication -> Users) if the created users are there.

### CODE REVIEW

**Review by bghoang — APPROVED** [review]
look good, standard auth with Firebase but it seems there is a conflict in package.json file

**Review by p8gonzal — COMMENTED** [review]
Looks good, will hold off on merge until we can merge backend and front end into one feature branch. Will also look into CI/CD options for the FirebaseConfig which contains the API key. Just nitpicking but if you could look into `install ci` to resolve package-lock.json conflicts

**Comment by p8gonzal** [comment]
Has been manually merged into experimental.

**Pull request closed without merging**
